Elaborate on some details of building and running a node. (#13115)
MinaProtocol/mina: README-dev.md

## Building Mina

> Building Mina is involved because many C library dependencies must be present in the system. Furthermore, these libraries need to be in correct versions, or else the system will fail to build. OCaml-specific setup is also required. Therefore, it is recommended to build Mina with Nix, which offers a great help in managing these dependencies. Manual dependency management is fragile and prone to break with every system update.

Reviewer: ```suggestion
Building Mina is involved because many C library dependencies must be present in the system. Furthermore, these libraries must be in the correct versions, or the system will fail to build. OCaml-specific setup is also required. We recommend building Mina with Nix, a functional package manager. Nix ensures that package dependency specifications are complete. Manual dependency management is fragile and prone to break with every system update.
```

Reviewer: @Sventimir 
hmmm we can probably omit  "Manual dependency management is fragile and prone to break with every system update."

Author: I find it extremely displeasing when others tell me what to do and so I try to avoid doing it myself. Instead I prefer to advise and recommend things, explaining why I think they're better to their alternatives, but ultimately leaving the decision to a well-informed user. I understand that this style complicates the message, though, while people often prefer simplicity over having very detailed information. Indeed, they often try to minimise decision-making when unnecessary.

On another note, I believe it's good to keep documentation impersonal wherever possible. It gives the reader the feeling they read about facts rather than oppinions of a group of people. It's somewhat related to the opinion I expressed above about presenting facts rather than opinions and letting people decide for themselves. Furthermore, it's usually unclear, who exactly that group expressiong their opinions in the documentation is (unless directly specified).

That said, I believe consistency is more important than any of these considerations, so if you think that simplified and/or personal style is better, I'm happy to accept that.

Reviewer: @Sventimir  thank you for your comment, I applaud your passion and interest. 
There are many considerations (and different opinions) for technical content, especially for developer docs. 
In this case, the reader of these README docs is the developer, so using second-person pronoun "you" is an established convention that we use for the Mina docs to write for our developer. 
Did you know we have a technical content style guide? https://github.com/o1-labs/docs2/wiki/Docs-Style-Guide#language-and-grammar 
You can always ask questions in the #mf-o1-docs Slack channel, too, so we can elevate our collective wisdom regarding docs (I love docs!)

Author: Thank you. I accept your proposals then.

The `daemon.json` config file also contains bootstrap data that is specific to the network the node is trying to connect to and must be tailored specifically for a particular network. This file can also override some of the configuration options selected during compilation. The `daemon.json` file can be extracted from the Docker image
that is dedicated to running a particular network. If it's not located in the `config` directory, it can be pointed to with `--config-file` option.

> The aforementioned bootstrap data includes the genesis ledger, i.e. the initial state of the blockchain. It is crucial for all the nodes on the network to have the same genesis ledger. While starting a new network, it is important that it contains at least one account possessing some funds. Otherwise, the network will not be able to bootstrap, as there will be no way to determine the next block producer.

Reviewer: ```suggestion
The previously mentioned bootstrap data includes the genesis ledger, which is the initial state of the blockchain. All the nodes on the network must have the same genesis ledger. A new network must contain at least one account with some funds before it can be started. If the network cannot determine the next block producer, it will not be able to bootstrap. 
```

Reviewer: r/w for clarity
